### PR TITLE
Reuse "viewport" option of autoresizing for export

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: ONE-5060
Preferred way how to turn on autoresizing of columns is "autoresizeAll" and old "viewport" will be now used during export to optimize performance and will autoresize visible columns only

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4063/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5697/ : FAILURE in 16m 15s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/992/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4064/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/992/ : UNSTABLE in 20m 09s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4068/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5700/ : FAILURE in 16m 31s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/995/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4072/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/995/ : SUCCESS in 20m 39s

**Comment by ivan-nejezchleb** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1455

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4076/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/202/ : SUCCESS in 10m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4075/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5704/ : SUCCESS in 16m 14s

File: libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts [hunk 1]
```diff
@@ -734,87 +738,103 @@ function valueFormatter(text: string, col: SeriesCol, separators: any) {
...
-
+    if (context) {
+        let calculatedColumnsTotalWidth = 0;
         config.columns.forEach((column: Column) => {
             const colDef: ColDef = column.getColDef();
             const colId = agColId(colDef);
             const maxWidth = colId ? maxWidths.get(colId) : undefined;
-            const possibleMaxWidth = getMaxWidth(context, colDef.headerName, !!colDef.sort, maxWidth);
+            if (shouldStopCalculation(config, calculatedColumnsTotalWidth)) {
+                return;
```
Comment (kandl) [inline libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts:810]: Nitpick: forEach can be probably replaced with for cycle, so it's not necessary to iterate through the rest of the columns after reaching clientWidth

File: libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts [hunk 2]
```diff
@@ -872,22 +892,26 @@ function getTableFonts(containerRef: HTMLDivElement): {
...
-            await new Promise((resolve) => {
+            await new Promise<void>((resolve) => {
                 setTimeout(() => {
                     ch.forEach((column: Column) => {
                         const columnDef = column.getColDef();
                         const colId = agColId(columnDef);
                         const autoResizedColumn = autoResizedColumns[colId];
 
                         if (colId && autoResizedColumn && autoResizedColumn.width) {
+                            // TODO After upgrade of aggrid to 23.1 or newer consider usage of columnApi.setColumnWidths with improved performance
```
Comment (kandl) [inline libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts:914]: Do we have a task for it? Can we mention the ticket number in the TODO?

File: libs/sdk-ui-tests/scenarios/pivotTable/autoresize.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default scenariosFor<IPivotTableProps>("PivotTable", PivotTable)
         ...PivotTableWithSingleColumn,
         config: {
             columnSizing: {
-                defaultWidth: "viewport",
+                defaultWidth: "autoresizeAll",
```
Comment (kandl) [inline libs/sdk-ui-tests/scenarios/pivotTable/autoresize.tsx:25]: Do we have tests for `viewport` option?

Comment (ivan-nejezchleb) [reply libs/sdk-ui-tests/scenarios/pivotTable/autoresize.tsx:25]: yes, here https://github.com/gooddata/gooddata-ui-sdk/pull/1455/files#diff-5586818ecff5be915a401c34cf1830a898d621c473c142845de6b665c8ab0928R116

File: libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts:914]: Created the ticket and mentioned in comment

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1455

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1182/ : SUCCESS in 1m 38s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1456/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/205/ : SUCCESS in 10m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5710/ : SUCCESS in 15m 47s

**Pull request merged**